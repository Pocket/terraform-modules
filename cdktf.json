{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
