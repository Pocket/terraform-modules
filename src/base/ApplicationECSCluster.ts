import { EcsCluster } from '@cdktf/provider-aws/lib/ecs-cluster';
import { TerraformMetaArguments } from 'cdktf';
import { Construct } from 'constructs';

export interface ApplicationECSClusterProps extends TerraformMetaArguments {
  prefix: string;
  tags?: { [key: string]: string };
}

/**
 * Generates an Application Certificate given a domain name and zoneId
 */
export class ApplicationECSCluster extends Construct {
  public readonly cluster: EcsCluster;

  constructor(
    scope: Construct,
    name: string,
    config: ApplicationECSClusterProps,
  ) {
    super(scope, name);

    this.cluster = new EcsCluster(this, `ecs_cluster`, {
      tags: config.tags,
      name: config.prefix,
      setting: [
        {
          name: 'containerInsights',
          value: 'enabled',
        },
      ],
      provider: config.provider,
    });
  }
}
