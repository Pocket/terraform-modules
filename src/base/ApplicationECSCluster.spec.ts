import { Testing, TerraformStack } from 'cdktf';
import { ApplicationECSCluster } from './ApplicationECSCluster';

test('renders an ECS cluster without tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationECSCluster(stack, 'testECSCluster', {
    prefix: 'bowling-',
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders an ECS cluster with tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationECSCluster(stack, 'testECSCluster', {
    prefix: 'bowling-',
    tags: {
      name: 'maude',
      description: 'artist',
    },
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});
