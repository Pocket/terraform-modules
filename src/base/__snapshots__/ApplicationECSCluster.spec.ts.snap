// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an ECS cluster with tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_ecs_cluster\\": {
      \\"test_testECSCluster_ecscluster_97BA6072\\": {
        \\"name\\": \\"bowling-\\",
        \\"tags\\": {
          \\"name\\": \\"maude\\",
          \\"description\\": \\"artist\\"
        },
        \\"setting\\": [
          {
            \\"name\\": \\"containerInsights\\",
            \\"value\\": \\"ENABLED\\"
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSCluster/ecs_cluster\\",
            \\"uniqueId\\": \\"test_testECSCluster_ecscluster_97BA6072\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders an ECS cluster without tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_ecs_cluster\\": {
      \\"test_testECSCluster_ecscluster_97BA6072\\": {
        \\"name\\": \\"bowling-\\",
        \\"setting\\": [
          {
            \\"name\\": \\"containerInsights\\",
            \\"value\\": \\"ENABLED\\"
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testECSCluster/ecs_cluster\\",
            \\"uniqueId\\": \\"test_testECSCluster_ecscluster_97BA6072\\"
          }
        }
      }
    }
  }
}"
`;
